File: maximumSubarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = Integer.MIN_VALUE;
        if (nums.length == 1){
            return nums[0];
        }
        for (int i = nums.length -2; i >= 0; i--){
            if (nums[i] + nums[i+1] > nums[i] ){
                nums[i] = nums[i+1] + nums[i];
            }
        }
        
         for (int i = 0; i < nums.length; i++){
            if (nums[i] > max){
                max = nums[i];
            }
        }

        return max;
        
    }
}

File: maximumSubarray.py
class Solution(object):
    def maxSubArray(self, nums):
        if len(nums) == 1:
            return nums[0]
        max = nums[len(nums) -1 ];
        # range( inclusive, exclusive)
        for i in range(len(nums)-2,-1,-1):
            if ( nums[i] < nums[i] + nums[i+1]):
                nums [i] = nums[i] + nums[i+1]
            if (nums[i] > max):
                max = nums[i]
        return max
        """
        :type nums: List[int]
        :rtype: int
        """
        
